aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -462,13 +460,19 @@ namespace {
           // might as well merge the CUDNN ones into a single tensor as well
           int mat_numel = *filter_dim_a.prod(at::ScalarType::Int).data_ptr<int>();
           if (linear_id == 0 || linear_id == num_linear_layers / 2) {
-            std::initializer_list<int64_t> size = {
-              mat_numel * num_linear_layers / 2, 1};
-            // Generate a new parameter tensor which is a view into the
-            // weight_buf.
-            Tensor param = at::empty({0}, weight_buf.options()).set_(weight_buf.storage(), offset, size);
-            params.emplace_back(std::move(param));
-            layer_params_count++;
+            // We could also exclude bias params by restricting cudnn_methods to just { cudnnGetRNNLinLayerMatrixParams }
+            // at the very top.  However, to do so would throw off the cur_offset account, which is currently a strict
+            // and informative check that all params are laid out the way we think they are.  If include_bias is false,
+            // I'd rather keep full cur_offset checks rather than save some CPU overhead by skipping the cudnn_method =
+            // cudnnGetRNNLinLayerBiasParams iteration.
+            if (include_bias || cudnn_method == cudnnGetRNNLinLayerMatrixParams) {
```

Comment:
This condition seems a bit backwards. Wouldn't `include_bias || cudnn_method != cudnnGetRNNLinLayerBiasParams` be more direct?

Reply:
works for me.

Reply:
hopefully resolved